### PR TITLE
Create a tunneldigger package for freifunk-berlin

### PR DESCRIPTION
**Pull request by pmelange**
Create the freifunk-berlin-tunneldigger package

This package is based on the gluon tunneldigger package.
https://github.com/freifunk-gluon/packages/tree/master/net/tunneldigger

The source for the tunneldigger client is at
https://github.com/wlanslovenija/tunneldigger

The differences to the gluon version are:
* The init script is disabled by default
* Fixed problems with starting multiple tunneldigger instances
* Added a hotplug script which waits for the bind_interface to go up/down before starting/stopping any associated tunneldigger
* Does not start tunneldigger for an interface if the interface is disabled
* Does not start tunneldigger if it has already been started
* Removes pid files when tunneldigger is stopped

Additionally all changes to uplink-tunnelberlin-tunneldigger and bbbdigger have been included

The init script is disabled in the luci wizard when tunneldigger is installed in the image.  If tunneldigger is installed via opkg (for example with bbbdigger) then the init script is disabled with the postinst script

### CODE REVIEW

**Comment by SvenRoederer** [comment]
can your changes not integrated into the current tunneldigger-package from gluon? 
Does it really make sense to create another package, with a lot of duplicated code? Having our own package needs a lot of maintenance. Having a common package for gluon and freifunk-berlin in save some ressources.

**Comment by pmelange** [comment]
The gluon package is just a Makefile, an example config file, and an init script.

Ours uses a similar Makefile, an example config file, and a totally different init script.  Additionally we use hotplug.

I really think it makes sense to diverge from how gluon is handling tunneldigger.

But all of that is stated above already.  I also find this easier to work with than maintaining a patch.

**Comment by pmelange** [comment]
@SvenRoederer I have integrated your changes into this branch. Could you please review and merge into master?

File: addons/freifunk-berlin-tunneldigger/files/tunneldigger.init [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+# Comment out the START declaration to prevent the init file from
+# actually starting tunneldigger at startup.  Instead rely on the
+# hotplug scripts
+#START=90
+
+PIDPATH=/var/run
+tunnel_id=0
+
+tunnel_is_up() {
```
Comment (SvenRoederer) [inline addons/freifunk-berlin-tunneldigger/files/tunneldigger.init]: the indentication is broken on the following lines

**Comment by SvenRoederer** [comment]
@pmelange  after merging this, feel free to add cf433e82fead91bdfc8ab134e5068e4aaca63e4d to master via another PR

File: addons/freifunk-berlin-tunneldigger/files/tunneldigger.init (hunk 1 above)
Comment (pmelange) [reply addons/freifunk-berlin-tunneldigger/files/tunneldigger.init]: fixed with commit edfc1fa

File: addons/freifunk-berlin-tunneldigger/files/tunneldigger.hotplug [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	config_get limit_bw_down "$cfg" limit_bw_down
+	config_get hook_script "$cfg" hook_script
+	config_get bind_interface "$cfg" bind_interface
+	config_get broker_selection "$cfg" broker_selection
+
+	let tunnel_id++
+
+	[ $enabled -eq 0 ] && return
+        [ ! -z ${bind_interface} ] && \
+                [ "$INTERFACE" != "${bind_interface}" ] && \
```
Comment (SvenRoederer) [inline addons/freifunk-berlin-tunneldigger/files/tunneldigger.hotplug]: will you fix indentication here also?

**Comment by SvenRoederer** [comment]
can you fixup the indentication-commits into 03cde0a8a250927ba27538125c74711469608b48 before merging?

**Comment by pmelange** [comment]
The indentation in the init and hotplug scripts are fixed.

**Comment by SvenRoederer** [comment]
quite huge commit. I assume it was a mixture of spaces and tabs used?

File: addons/freifunk-berlin-tunneldigger/files/tunneldigger.hotplug (hunk 2 above)
Comment (SvenRoederer) [reply addons/freifunk-berlin-tunneldigger/files/tunneldigger.hotplug]: ping

**Pull request merged**